### ISSUE TITLE
Feature Request: Resource for managing AWS Glue User Defined Functions

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#3883]
Split feature request from #1416. For issues not immediately being worked on, please use :+1: upvotes on this original issue comment to help guage community interest.

### Terraform Version

```
terraform 0.10+
terraform-provider-aws (new feature)
```

### Affected Resource(s)

- `aws_glue_user_defined_function`

### Terraform Configuration Files

```hcl
# New Resource - not implemented
resource "aws_glue_user_defined_function" "example" {
  # ...
}
```

### Expected Behavior

Create, read, update, and delete support for AWS Glue user defined functions.

### Actual Behavior

Resource not implemented.

### References

* https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-catalog-functions.html
* https://docs.aws.amazon.com/sdk-for-go/api/service/glue/#Glue.CreateUserDefinedFunction

**Comment by bflad** [comment]
This is currently WIP at #12537

**Comment by bflad** [comment]
Support for this new resource has been merged and will release with version 3.5.0 of the Terraform AWS Provider, later this week. Thanks to @DrFaust92 for the implementation. 👍
